[BUG] Navbar for some size is breaking a line

andriyParashchuk opened the pull request:
![Image](https://github.com/user-attachments/assets/d19a44cc-8f3c-4d1d-8585-1d95d6d1a35b)



<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
	- Added descriptive titles and detailed descriptions for various services and use cases to improve clarity and appeal.
	- Introduced a responsive navigation structure for enhanced user interaction across devices.
	- Enhanced mobile navigation experience with interactive elements for easier access to options.

- **Style**
	- Improved visual presentation and user interaction with new CSS styles, including responsive navigation and enhanced button styles.
	- Streamlined HTML structure by removing inline styles and incorporating external stylesheets for better maintainability.

- **Bug Fixes**
	- Simplified mobile menu functionality, improving usability and reducing complexity.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

pftg commented:
@AucT would you be able to handle this, at least for blog?

pftg commented on static/assets/css/style.css:
```diff
@@ -41,8 +25,12 @@ Appearance > Theme Settings > Code or create a child theme.
+  max-height: 75vh;
+  margin: 0 auto;
+  display: block;
 }
```
@andriyParashchuk is it related to navbar?

pftg replied:
@AucT could you add notes about responsive images and this style

andriyParashchuk replied:
@pftg nope. That appeared after rebasing

pftg commented on static/assets/css/style.css:
```diff
@@ -176,3 +164,177 @@ Appearance > Theme Settings > Code or create a child theme.
+  margin-bottom: 0;
+  margin-top: 5px;
+  color: #7e7e7e;
+}
```
@andriyParashchuk please extract to separate css all stuff related to navbar

pftg commented on themes/beaver/layouts/partials/page/header.html:
```diff
@@ -0,0 +1,151 @@
+    <body class="privacy-policy page-template-default page page-id-3059 fl-builder fl-theme-builder-header fl-theme-builder-header-header fl-theme-builder-footer fl-theme-builder-footer-footer fl-framework-base-4 fl-preset-default fl-full-width fl-search-active" itemscope="itemscope" itemtype="https://schema.org/WebPage">
+      <div class="fl-page">
+
+        {{ partial "page/top-panel.html" . }}
```
instead of adding new files, please update `baseof.html` and replace appropriate `header` tag with `{{ partial "page/top-panel.html" . }}`

pftg reviewed (changes requested):
Need some changes

andriyParashchuk committed:
#53: Extract navigation to partial

andriyParashchuk committed:
refact: top navigation to reduce css and simplify html

 #53: Extract navigation to partial

 #53: Cleanup

 #53: Add mobile menu

 #53: Cleanup

 #53: Rewor to js

 #53: Resolve critical path conflicts

lint: reformatted

 fix: loading of assets

fix: js

fix: open menu

refact: load navigation for everybody

andriyParashchuk committed:
#53: Cleanup

andriyParashchuk committed:
#53: Add pixel perfect

andriyParashchuk committed:
#53: Fix define active link

andriyParashchuk committed:
#53: Add mobile opener

andriyParashchuk committed:
#53: Add mobile menu

andriyParashchuk committed:
#53: Rebase branch

andriyParashchuk committed:
#53: Cleanup

andriyParashchuk committed:
#53: Remove unused style

andriyParashchuk committed:
#53: Re-connect styles and scripts

andriyParashchuk committed:
#53: Rewor to js

andriyParashchuk committed:
#53: Fix mobile menu script

pftg commented on themes/beaver/layouts/partials/page/site-scripts.html:
```diff
@@ -1,12 +1,10 @@
 $js := slice
   (resources.Get "js/587.js")
   (resources.Get "js/bb-theme/pp_advanced_tabs.js")
+  (resources.Get "js/navigation.js")
```
make it as `js/bb-theme/pp_advanced_menu.js` as separate js, because it's critical and will be everywhere

pftg replied:
```html
<script defer
        src='{{ (resources.Get "js/navigation.js" | js.Build (dict "minify" true) | minify | fingerprint | resources.PostProcess).RelPermalink }}'>
</script>
```

pftg replied:
but it's maybe ok for now

pftg commented on themes/beaver/layouts/partials/page/header.html:
```diff
@@ -1,123 +1,11 @@
-            </div>
-        </div>
+<header class="top-panel">
+  <div class="container">
```
Is this bootstrap's container?

andriyParashchuk replied:
Hard to say. Critical css make unclear that 
Let's rename

pftg reviewed (changes requested):
1. Mobile menu toggler is not working 
![Accelerate   Achieve Your Product Vision 2024-09-16 12-54-18](https://github.com/user-attachments/assets/c6afe7ba-70b1-4eb3-aba4-eb01df636575)

2. sub-menu fonts is bold: head and icons are bolder then on the site

pftg committed:
Merge branch 'master' into 53-fix-top-navigation

pftg commented:
@andriyParashchuk please check 404 page as well 
![jetthoughts github io – Commit: 404 png 2024-09-16 23-12-34](https://github.com/user-attachments/assets/ecb50d8b-ef5b-4a37-a066-6411b86ee356)

andriyParashchuk committed:
#53: Resolve critical path conflicts

andriyParashchuk committed:
#53: Fix identical spacing

andriyParashchuk commented:
@pftg 
1. Menu fixed. Problem was in critical css
2. font-weight and font-family in inspection looks like the same. However you are right - that looks more bolder. Need to check
3. Yeah. 404 has own life. Need to fix

pftg commented:
- [ ] some styles still broken after rebase
- [ ] menu is not working for mobile

![Accelerate   Achieve Your Product Vision 2024-09-17 12-25-29](https://github.com/user-attachments/assets/c5a20a4a-c575-4a02-b507-a332ba58cf71)

![Uploading Accelerate & Achieve Your Product Vision 2024-09-17 12-27-43.jpg…]()

pftg commented:
@andriyParashchuk https://www.loom.com/share/f20b6c7eebed4a9da1265365c9fc667d and use cases are broken

pftg commented:
![Learn More About Us and Our Story JetThoughts 2024-09-17 12-43-01](https://github.com/user-attachments/assets/65c727c5-fb3e-4186-b3b8-de5024999545)

andriyParashchuk committed:
#53: Rebase

andriyParashchuk committed:
#53: Rebase

andriyParashchuk committed:
#53: Fix icons path

andriyParashchuk committed:
fixes buttons and menu

use navCss as assets

fix nav wrapper

remove duplicate line

remove not needed file

andriyParashchuk commented:
@pftg please review

pftg commented:
@AucT please check locally and create a screencast to prove does it work r not

pftg commented on themes/beaver/layouts/_default/baseof.html:
```diff
@@ -63,7 +63,7 @@
     <meta name="msapplication-config" content="{{ (resources.Get "img/favicon/browserconfig.xml" | resources.ExecuteAsTemplate "browserconfig.xml" . | fingerprint).RelPermalink }}">
     <meta name="theme-color" content="#ffffff">
-
+    <link rel='stylesheet' href='{{ absURL "assets/css/navigation.css" }}' media='all' />
```
@AucT is it correct line?

AucT replied:
No, if you want use assets with fingerprint
Yes, if you want use static `navigation.css` from `static/assets/css/navigation.css`

I see navgation.css in assets, then it is not correct.

Here code to use

```
{{ $navCss := resources.Get "css/navigation.css" | minify | fingerprint }}
<link rel="stylesheet" href="{{ $navCss.RelPermalink }}">
```

pftg replied:
@AucT exactly, please give @andriyParashchuk explicit recommendation how it should be, based on our last expectations

pftg commented on themes/beaver/layouts/page/index.html:
```diff
@@ -1,1157 +1,1157 @@
+{{ partial "header/critical/homepage.html" . }}
+
+{{ $CSS := slice
+   (resources.Get "css/components/pp_adv_menu.css")
```
@AucT what about this?

AucT commented:
![1ftHIOFcOv](https://github.com/user-attachments/assets/e3dad332-cbb5-4543-98c8-ce77be3febcf)
![unsar73c4o](https://github.com/user-attachments/assets/077e43fc-04f9-4655-a897-dfbf5bb0fff4)

pftg commented:
Guys, the desktop menu has a hidden

![image](https://github.com/user-attachments/assets/dc40010c-f6ee-4dff-8135-074a15fa49c2)

pftg reviewed (changes requested):
@AucT please help @andriyParashchuk to handle last issue

pftg committed:
Merge branch 'master' into 53-fix-top-navigation

pftg committed:
test: updates screenshots
